### ISSUE TITLE
Not using the train_set?

### ISSUE THREAD

**Issue by Spiffical** [mesarcik/ROAD#22]
Hey there! I am using your code for a similar project (identifying anomalous spectrograms from underwater microphones), and I am getting pretty good results out of the box. I noticed, however, that the train_set in the .h5 file is never getting used (the _join() function stitches together the test_data and anomaly_data only). Is there a reason for this?

**Comment by mesarcik** [comment]
Sorry for the confusion, the terms "test" and "training" are used quite liberally in the codebase, it comes down to the fact that we are training 2 different models with different training objectives and different data requirements.  You correctly point out that the `_join()` function is used to add **all** anomalous samples to the "test" dataset. In the case of our training paradigm we train 2 different models, the SSL model and the supervised classifier, and two different datasets are used for training them. 

In the case of the supervised classifier, we train on half of the normal data (test set) and half of the anomalous data; the training set is not used as it comprises only of "normal" data. Whereas for the SSL classifier, we want only to train on "normal" data, such that the model only learn representations of normality (so here the "training set" is used). Then for fine-tuning, we use the anomalous data for fine tuning. We show in the paper that this approach seems to give improved robustness, classification performance, etc (see Figure 5: https://arxiv.org/pdf/2307.01054)  over the supervised classifier. 

You could technically try use the rest of the normal "training data", but im not sure if it would yield better classification performance for the supervised classifier.
